lib/Dialect/LLHD/Simulator/Trace.cpp

```diff
@@ -33,7 +33,7 @@ Trace::Trace(std::unique_ptr<State> const &state, llvm::raw_ostream &out,
 // Changes gathering methods
 //===----------------------------------------------------------------------===//
 
-void Trace::pushChange(std::string inst, unsigned sigIndex, int elem = -1) {
+void Trace::pushChange(unsigned inst, unsigned sigIndex, int elem = -1) {
```

Comment:
Very good to see all these integer IDs instead of strings. I wonder if it would make sense to wrap these in a struct, or at least a typedef, to make their use clear? Something like `InstId`?

Reply:
So something along the lines of 
```
struct SignalLocation {
  unsigned instIndex;
  unsigned sigIndex;
  int elem = -1;
}
```
and then only have a `SignalLocation` as input?

Reply:
Yeah if we generally pair instances, signals, and elements up like this. But I've seen other places where we use just instance indices. So I think it would make sense to introduce a newtype for the instance ID, to make it clear when you read the code that this is not just any `unsigned`, but it has the semantics of a `InstId`. Just nitpicking though, that's definitely not an urgent/critical change 😛

Reply:
Mhh having it as a "different" type sounds a bit strange to me, as it's pretty much just index. Am I missing something? Would also just renaming the variable to `instIndex` (more similar to the signal index) or `instID` not make it already more clear?